mitodl/ocw_oer_export
# Create OER Commons CSV from OCW

# What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/3147

# Description (What does it do?)
This PR is responsible for the business logic necessary for creating the CSV out of MIT Open API (OCW) data according to OER Commons Template.

This will work as a PyPI package as well as CLI interface. 

Please refer to the instructions in `README.md` for more details.


# How can this be tested?
1. Checkout to this branch
2. Create Virtual Environment and activate it -- something like `python -m venv venv`
3. Make a build using `python -m build`. If you get an error relating to `module build not found`, install it using: `python -m pip install build` in your currently activated `venv`
4. After you have made the build, run: `pip install --force-reinstall dist/ocw_oer_export-1.0.0-py3-none-any.whl`
5. Now you can use the project as a python package. For CLI, run the command: `ocw-oer-export --create_csv`. This should create a CSV in the current working directory.

To use it as an import in another project, instructions are same:
1. Create an empty python test project
2. Create & activate Virtual Environment
3. Place the same file `ocw_oer_export-1.0.0-py3-none-any.whl` in the project directory
4. Run: `pip install --force-reinstall ocw_oer_export-1.0.0-py3-none-any.whl`
5. Now you can start using it in your files:
```
from ocw_oer_export import create_json

create_csv()

```
6. Run using: `python test.py`
